### PR TITLE
Removed parallel restore from ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress testcase

### PR DESCRIPTION
**Pull request by snigdha-px**
**What this PR does / why we need it**:
Removed parallel restore from ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress testcase
This is to make sure there are no parallel restores happening
**Which issue(s) this PR fixes** (optional)
Closes #https://portworx.atlassian.net/browse/PB-6255

**Special notes for your reviewer**:
https://aetos.pwx.purestorage.com/resultSet/testSetID/584184/
https://jenkins.pwx.dev.purestorage.com/job/portworx-backup/job/system-tests/job/byoc/job/px-backup-on-demand-system-test-byoc/4893/

### CODE REVIEW

**Review by kphalgun-px — APPROVED** [review]
LGTM

**Review by ak-px — APPROVED** [review]
LGTM

**Review by vpinisetti-px — APPROVED** [review]
LGTM

**Review by kshithijiyer-px — APPROVED** [review]

File: tests/backup/backup_resiliency_test.go [hunk 1]
```diff
@@ -1517,17 +1517,10 @@ var _ = Describe("{ScaleDownPxBackupPodWhileBackupAndRestoreIsInProgress}", func
...
-					defer GinkgoRecover()
-					defer wg.Done()
-					defer func() { <-sem }()
-					_, err = CreateRestoreWithoutCheck(restoreName, backupName, nil, DestinationClusterName, BackupOrgID, ctx)
-					dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
-				}(backupName)
+				_, err = CreateRestoreWithoutCheck(restoreName, backupName, nil, DestinationClusterName, BackupOrgID, ctx)
+				dash.VerifyFatal(err, nil, fmt.Sprintf("Restoring the backup %s with name %s", backupName, restoreName))
 			}
 			wg.Wait()
```
Comment (mkoppal-px) [inline tests/backup/backup_resiliency_test.go:1525]: Why do we need wait here?

Comment (snigdha-px) [reply tests/backup/backup_resiliency_test.go]: Missed to remove.
Removed now

**Review by mkoppal-px — APPROVED** [review]
LGTM

**Pull request merged**